src/settings/settings.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/sequelize';
+import { CreateSettingDto } from './dto/create-setting.dto';
+import { UpdateSettingDto } from './dto/update-setting.dto';
+import Setting from '../models/setting.model';
+
+@Injectable()
+export class SettingsService {
+  constructor(
+    @InjectModel(Setting)
+    private settingModel: typeof Setting,
+  ) {}
+
+  async create({ name, account_id, data_type, value }: CreateSettingDto) {
+    return this.settingModel.create({ name, account_id, data_type, value });
```

Comment:
Creating a setting with `dataType` object/boolean would fail if we don't convert them to string here

Reply:
This is implemented in [line 34](https://github.com/ananth243/nest-starter-project/blob/385cc411a4493fac8e41fec9fd2c6c8ba0ce4c33/src/settings/settings.validation.ts#L34) of settings.validation.ts

Reply:
Shifted logic to [service level](https://github.com/ananth243/nest-starter-project/blob/732f3cf7c7bc7cf884500fc5eda0ff5aa6709c60/src/settings/settings.service.ts#L24)